Orange/widgets/widget.py

```diff
@@ -329,11 +329,27 @@ def setStatusBarText(self, text, timeout=5000):
 
     def __restoreWidgetGeometry(self):
         restored = False
-        if self.save_position:
+        # Do not attempt to restore/set geometry when the widget is already in
+        # full screen/maximized state (also minimized?). More often than not
+        # this results in a borked window state.
+        if self.save_position and \
+                not self.windowState() & (Qt.WindowFullScreen | Qt.WindowMaximized):
             geometry = self.savedWidgetGeometry
             if geometry is not None:
+                # strip full screen state from the stored geometry if presetn
+                try:
+                    state = geom_state.from_bytes(geometry)
+                except ValueError:
+                    pass
+                else:
+                    if state.full_screen:
+                        # remove the full screen state replacing it with
+                        # maximized state instead
+                        state = state._replace(
+                            full_screen=0, maximized=Qt.WindowMaximized
+                        )
+                        geometry = state.to_bytes()
                 restored = self.restoreGeometry(QByteArray(geometry))
```

Comment:
The other two changed files above surely must raise the question:

Why didn't you settle with something like:
```py
self.setWindowState(self.windowState() & ~Qt.WindowFullScreen)
geom, avail = self.geometry(), qApp.desktop().availableGeometry(self)
if geom.width() > avail.width() and geom.height() > avail.height():
    self.setWindowState(self.windowState() | Qt.WindowMaximized)
```
The docs say:
> If the window is not visible, the window state will take effect when show() is called. For visible windows, the change is immediate.